pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignmentUtils.java

```diff
@@ -87,6 +87,15 @@ private SegmentAssignmentUtils() {
     return instances;
   }
 
+  static List<String> getInstancesForDimTable(InstancePartitions instancePartitions) {
```

Comment:
We don't need instance partitions for dimTable. It should be assigned to all the instances under the given tag

Reply:
Removed this way of gathering hosts, now explicitly returning all the hosts under a given tag as you suggested